### PR TITLE
feat: filters associated systems by groups' ids and names

### PR DESCRIPTION
**Pull request by petrblaho**
ESSNTL-5276

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: system_baseline/models.py [hunk 1]
```diff
@@ -59,18 +59,23 @@ def mapped_system_ids(self, rbac_group_filters=None):
         # [{'id': None}, {'id': '39eb2f52-37c1-4f75-95e6-7237d0b385b7'}]
 
         mapped_systems_query = self.mapped_systems
+
+        def _get_query_from_filters(filters):
+            return [
+                SystemBaselineMappedSystem.groups == "[]"
+                if "id" in rbac_group_filter and rbac_group_filter["id"] is None
+                else SystemBaselineMappedSystem.groups.contains(cast([rbac_group_filter], JSONB))
+                for rbac_group_filter in filters
```
Comment (vkrizan) [inline system_baseline/models.py:68]: `rbac_group_filter` looks similar to `rbac_group_filters` from outer function.

Comment (petrblaho) [reply system_baseline/models.py:68]: This is the remnant of refactoring and moving the code. Thank you!

File: system_baseline/models.py [hunk 2]
```diff
@@ -48,7 +48,15 @@ def validate_facts(self, key, value):
         validators.check_for_duplicate_names(value)
         return value
 
-    def mapped_system_ids(self, rbac_group_filters=None):
+    def get_query_from_filters(self, filters):
```
Comment (vkrizan) [inline system_baseline/models.py:51]: Name suggestion: `get_groups_query_filter`

**Review by vkrizan — APPROVED** [review]

File: system_baseline/models.py (hunk 2 above)
Comment (petrblaho) [reply system_baseline/models.py]: Naming is hard...

**Pull request merged**